lib/table_of_contents/parser.rb

```diff
@@ -17,13 +23,24 @@ def toc
       end
 
       def build_toc
-        %(<ul class="#{@configuration.list_class}">\n#{build_toc_list(@entries)}</ul>)
+        %(<ul id="toc" class="#{@configuration.list_class}">\n#{build_toc_list(@entries)}</ul>)
       end
 
       def inject_anchors_into_html
         @entries.each do |entry|
-          entry[:header_content].add_previous_sibling(
-            %(<a class="anchor" href="##{entry[:id]}" aria-hidden="true"><span class="octicon octicon-link"></span></a>)
+          # Add id to h-element
+          entry[:header_parent].set_attribute('id', "#{entry[:id]}.to_s")
+
+          # Add link icon after text
+          entry[:header_content].add_next_sibling(
+            %(<a class="anchor" href="##{entry[:id]}" aria-hidden="true">&nbsp;&#128279;</a>)
+          )
+
+          # Add link 'nav to toc'
+          arr_to_top = [2, 3]
+          next unless arr_to_top.include?(entry[:h_num])
+          entry[:header_content].add_next_sibling(
+            %(<span style="float: right"><a class="anchor_to_top" href="#toc" aria-hidden="true">&#x21A5;</a></span>)
```

Comment:
I'm wondering why you need this navigation. 🤔 

I'd like to know the use case. Could you give me screen capture?

Reply:
[Screenshot of Navigate to TOC in OneDrive](https://1drv.ms/u/s!Ata8w9T7BmKKkIAG7yKLjb90ojtAWA?e=KZt1D3)

At the end of the headline line (float: right) I placed an arrow symbol / Unicode char. OnClick the user will jump to the beginning of the toc, alternative it could be possible to jump to the beginning of the page (top).